demosplan/DemosPlanCoreBundle/Security/Authentication/Provider/SecurityUserProvider.php

```diff
@@ -36,33 +35,37 @@ public function refreshUser(UserInterface $user): ?UserInterface
             throw new UnsupportedUserException(sprintf('Invalid user class %s', $user::class));
         }
 
-        return new SecurityUser($this->loadUserByLogin($user->getUsername()));
+        // Return User object here as we want to have the User object in the Session instead,
+        // as the SecurityUser is only meant to be used during Authentication.
+        // In {@DemosPlanResponseListener::transformTokenUserObjectToSecurityUserObject()} we
+        // transform the User object to a SecurityUser object to save it between requests.
+        return $this->loadUserByLogin($user->getUsername());
     }
 
     public function supportsClass(string $class): bool
     {
         return SecurityUser::class === $class;
     }
 
-    public function loadUserByUsername(string $username): SecurityUser
+    public function loadUserByUsername(string $username): User
     {
         return $this->loadUserByIdentifier($username);
     }
 
-    public function loadUserByIdentifier(string $identifier): SecurityUser
+    public function loadUserByIdentifier(string $identifier): User
+    {
+        return $this->loadUserByLogin($identifier);
+    }
+
+    public function getSecurityUser(string $identifier): SecurityUser
     {
         return new SecurityUser($this->loadUserByLogin($identifier));
     }
 
-    public function upgradePassword(UserInterface $user, string $newEncodedPassword): void
+    public function upgradePassword(UserInterface $user, string $newHashedPassword): void
     {
         $userEntity = $this->loadUserByLogin($user->getUsername());
-        // set the new encoded password on the User object
-        $userEntity->setPassword($newEncodedPassword);
-        $userEntity->setAlternativeLoginPassword($newEncodedPassword);
-
-        // execute the queries on the database
-        $this->entityManager->flush();
+        $this->userRepository->upgradePassword($userEntity, $newHashedPassword);
```

Comment:
IDE says "unhandled exceptions".
But it seems that the named exceptions (in the docblock) of the used `upgradePassword()` are incorrect.

Reply:
indeed dbd7b6a2